File: app/src/main/java/ua/com/foxminded/locationtrackera/mvi/states/AbstractEffect.java
package ua.com.foxminded.locationtrackera.mvi.states;

public abstract class AbstractEffect<T> extends ScreenEffect<T> {

    private boolean isHandled = false;

    @Override
    public void visit(T screen) {
        if (!isHandled) {
            handle(screen);
            isHandled = true;
        }
    }

    public abstract void handle(T screen);
}

File: mvi/src/main/java/ua/com/foxminded/locationtrackera/mvi/states/AbstractEffect.kt
package ua.com.foxminded.locationtrackera.mvi.states

abstract class AbstractEffect<T> : ScreenEffect<T>() {
    private var isHandled = false

    override fun visit(screen: T) {
        if (!isHandled) {
            handle(screen)
            isHandled = true
        }
    }

    abstract fun handle(screen: T)
}
